src/django_components/template_parser.py

```diff
@@ -41,7 +44,7 @@
     num=r"[-+\.]?\d[\d\.e]*",
     # The following is the only difference from the original FilterExpression. We allow variable names to have extra
     # special characters: - : . @ #
-    var_chars=r"\w\-\:\@\.\#",
```

Comment:
Refactored the regex out, so we can re-use it to validate the user-generated tag names from TagFormatter